### ISSUE TITLE
Can not override the KafkaServer constructor

### ISSUE THREAD

**Issue by smhmayboudi** [nestjs/nest#5599]
## Current behavior

I want to rewrite KafkaServer. The option is private.

## Input Code

https://github.com/nestjs/nest/blob/master/packages/microservices/server/server-kafka.ts#L45

```ts
import {
  KAFKA_DEFAULT_CLIENT,
  KAFKA_DEFAULT_GROUP,
} from "@nestjs/microservices/constants";
import { KafkaOptions, ServerKafka } from "@nestjs/microservices";

import { KafkaConfig } from "@nestjs/microservices/external/kafka-options.interface";

export class AppServerKafka extends ServerKafka {
  constructor(protected readonly options: KafkaOptions["options"]) {
    super(options);
    const clientOptions =
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      this.getOptionsProp(this.options, "client") || ({} as KafkaConfig);
    this.clientId =
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
      (clientOptions.clientId || KAFKA_DEFAULT_CLIENT) + "_SERVER";
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
    this.groupId = (consumerOptions.groupId || KAFKA_DEFAULT_GROUP) + "_SERVER";
  }
}
```

## Expected behavior

It should be protected.

## Possible Solution

```ts
constructor(private readonly options: KafkaOptions['options']) {
```
should be change to
```ts
constructor(protected readonly options: KafkaOptions['options']) {
```
## Environment

<pre><code>
Nest version: 7.5.1

 
For Tooling issues:
- Node version: v12.19.0  
- Platform:  Mac

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by smhmayboudi** [comment]
Yes. Would you please provide me a guideline?

**Comment by kamilmysliwiec** [comment]
Fixed in 7.5.0
